src/data.ts

```diff
@@ -3,16 +3,21 @@
  */
 import {VgData} from './vega.schema';
 
+export interface Parse {
+  [field: string]: null | string | 'string' | 'boolean' | 'date' | 'number';
+}
+
 export interface DataFormatBase {
   /**
-   * If set to auto (the default), perform automatic type inference to determine the desired data types.
-   * Alternatively, a parsing directive object can be provided for explicit data types. Each property of the object corresponds to a field name, and the value to the desired data type (one of `"number"`, `"boolean"` or `"date"`).
+   * If set to `"auto"` (the default), perform automatic type inference to determine the desired data types.
+   * If set to `null`, disable type inference based on the spec and only use type inference based on the data.
+   * Alternatively, a parsing directive object can be provided for explicit data types. Each property of the object corresponds to a field name, and the value to the desired data type (one of `"number"`, `"boolean"`, `"date"`, or null (do not parse the field)).
    * For example, `"parse": {"modified_on": "date"}` parses the `modified_on` field in each input record a Date value.
    *
    * For `"date"`, we parse data based using Javascript's [`Date.parse()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse).
    * For Specific date formats can be provided (e.g., `{foo: 'date:"%m%d%Y"'}`), using the [d3-time-format syntax](https://github.com/d3/d3-time-format#locale_format). UTC date format parsing is supported similarly (e.g., `{foo: 'utc:"%m%d%Y"'}`). See more about [UTC time](timeunit.html#utc)
    */
-  parse?: 'auto' | object;
+  parse?: 'auto' | Parse | null;
```

Comment:
What's the behavior for  providing an object for only a subset of fields?

(Do we do "auto" for all other fields unmentioned?)